pkg/upgrade/main_test.go

```diff
@@ -0,0 +1,44 @@
+package upgrade
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVersions(t *testing.T) {
+	maptoTest := map[string]upgradeFunction{
+		"1.17.0": noop,
+		"1.15.4": noop,
+		"1.15.0": noop,
+		"1.16.1": noop,
+		"1.12.2": noop,
+	}
+	sortedSemVersions := []string{
+		"1.12.2", "1.15.0", "1.15.4", "1.16.1", "1.17.0",
+	}
+
+	semVersions, err := versions(maptoTest)
+	assert.NoError(t, err)
+	for i, v := range semVersions {
```

Comment:
Can't this just be a straight equality check between the arrays?

Reply:
They are not same type, but I did a change and now I just compare equality.

Reply:
Good point :)